clientcontroller/babylon.go

```diff
@@ -788,3 +804,110 @@ func (bc *BabylonController) QueryPublicRandCommitList(fpPk *btcec.PublicKey, st
 
 	return commitList, nil
 }
+
+// reliablySendMsgsResendingOnMsgErr sends the msgs to the chain, if some msg fails to execute
+// and contains 'message index: %d', it will remove that msg from the batch and send again
+// if there is no more message available, returns the last error.
+func (bc *BabylonController) reliablySendMsgsResendingOnMsgErr(
+	ctx context.Context,
+	msgs []sdk.Msg,
+	expectedErrs []*sdkErr.Error,
+	unrecoverableErrs []*sdkErr.Error,
+) (*types.TxResponse, error) {
+	var err error
+
+	maxRetries := BatchRetries(msgs, 10)
+	for i := uint64(0); i < maxRetries; i++ {
+		// Use testClient for testing, otherwise use bbnClient
+		var client babylonClient
+		if bc.testClient != nil {
+			client = bc.testClient
+		} else {
+			client = bc.bbnClient
+		}
+
+		res, errSendMsg := client.ReliablySendMsgs(ctx, msgs, nil, unrecoverableErrs)
```

Comment:
why do we put `nil` here instead of `expectedErrs` and handle `expectedErrs` separately in Line 834?